pkg/syncer/metadatasyncer.go

```diff
@@ -2909,3 +2915,80 @@ func startStoragePolicyQuotaCRInformer(ctx context.Context, cfg *restclient.Conf
 	}()
 	return nil
 }
+
+// storagePolicyUsageCRSync patches StoragePolicyUsage CRs for k8s PVs in Bound state
+func storagePolicyUsageCRSync(ctx context.Context, metadataSyncer *metadataSyncInformer) {
+	log := logger.GetLogger(ctx)
+	log.Debugf("Starting CnsVolumeOperationRequest CR sync.")
+	// Get K8s PVs in State "Bound".
+	k8sPVsInBoundState, err := getBoundPVs(ctx, metadataSyncer)
+	if err != nil {
+		log.Errorf("Failed to get PVs from kubernetes. Err: %+v", err)
+		return
+	}
+	k8sVolumesToNamespaceMap := make(map[string][]*v1.PersistentVolume)
+	for _, pv := range k8sPVsInBoundState {
+		k8sVolumesToNamespaceMap[pv.Spec.ClaimRef.Namespace] =
+			append(k8sVolumesToNamespaceMap[pv.Spec.ClaimRef.Namespace], pv.DeepCopy())
+	}
+	// Prepare Config and NewClientForGroup for cnsOperatorClient
+	restConfig, err := config.GetConfig()
+	if err != nil {
+		log.Errorf("failed to get Kubernetes config. Err: %+v", err)
+		return
+	}
+	cnsOperatorClient, err := k8s.NewClientForGroup(ctx, restConfig, cnsoperatorv1alpha1.GroupName)
+	if err != nil {
+		log.Errorf("Failed to create CnsOperator client. Err: %+v", err)
+		return
+	}
+	// Get the list of all StoragePolicyUsage CRs from all supervisor namespaces
+	storagePolicyUsageList := &storagepolicyusagev1alpha1.StoragePolicyUsageList{}
+	err = cnsOperatorClient.List(ctx, storagePolicyUsageList)
+	if err != nil {
+		log.Errorf("failed to list %s CR from all supervisor namespaces. Error: %+v",
+			storagepolicyusagev1alpha1.CRDSingular, err)
+		return
+	}
+	// Get the list of all CnsVolumeInfo CRs from vmware-system-csi namespace
+	namespace := common.GetCSINamespace()
+	cnsVolumeInfoList := &cnsvolumeinfov1alpha1.CNSVolumeInfoList{}
+	err = cnsOperatorClient.List(ctx, cnsVolumeInfoList, client.InNamespace(namespace))
+	if err != nil {
+		log.Errorf("failed to list %s CR from supervisor namespace %q. Error: %+v",
+			cnsvolumeinfov1alpha1.CRDSingular, namespace, err)
+		return
+	}
+	cnsVolumeInfoMap := make(map[string]*cnsvolumeinfov1alpha1.CNSVolumeInfo)
+	for _, volumeInfo := range cnsVolumeInfoList.Items {
+		cnsVolumeInfoMap[volumeInfo.Name] = volumeInfo.DeepCopy()
+	}
+	if len(cnsVolumeInfoList.Items) > 0 {
+		// Iterate through storagePolicyUsageList
+		for _, storagePolicyUsage := range storagePolicyUsageList.Items {
+			totalUsedQty := resource.NewQuantity(int64(0), resource.BinarySI)
+			// For every storagePolicyUsage, fetch "Bound" PVs in that namespace from k8sVolumesToNamespaceMap
+			if volumes, ok := k8sVolumesToNamespaceMap[storagePolicyUsage.Namespace]; ok {
+				for _, pv := range volumes {
+					// Verify the StorageClass, StoragePolicyId match with the storagePolicyUsage spec
+					// using the CnvOlumeInfo CR for the volume
```

Comment:
Typo:  CnvOlumeInfo should be "cnsVolumeInfo:

Reply:
Done